ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/KVCommandDeduplicationIT.scala

```diff
@@ -58,23 +57,40 @@ final class KVCommandDeduplicationIT(timeoutScaleFactor: Double, ledgerTimeInter
           }
         }
         for {
-          time <- ledger.time()
+          time <- anyParticipant.time()
           updatedModel = timeModelUpdate(timeModel.getTimeModel)
-          timeModelForTest <- ledger
-            .setTimeModel(
+          (timeModelForTest, participantThatDidTheUpdate) <- tryUpdateOnAllParticipants(
+            participants,
+            _.setTimeModel(
               time.plusSeconds(30),
               timeModel.configurationGeneration,
               updatedModel,
             )
-            .map(_ => updatedModel)
-            .recover { case NonFatal(e) =>
-              logger.warn("Failed to update time model, running with default", e)
-              timeModelIfUpdateFailed
-            }
+              .map(_ => updatedModel),
+          )
           _ <- test(timeModelForTest)
-            .transformWith(testResult => restoreTimeModel().transform(_ => testResult))
+            .transformWith(testResult =>
+              restoreTimeModel(participantThatDidTheUpdate).transform(_ => testResult)
+            )
         } yield {}
       })
   }
 
+  /** Try to run the [[update]] sequentially on all the participants.
+    * The function returns the first success or the last failure of the update operation.
+    * Useful for updating the configuration when we don't know which participant can update the config, as only one has the permissions to do so.
+    */
+  private def tryUpdateOnAllParticipants(
```

Comment:
I would name it `tryUpdateTimeModel`, as it contains the `TimeModel`. If you'd like to keep the name, I would make it generic. However, not sure if worth it, if there is just one usage. Always happy to bikeshed.

Reply:
```suggestion
  private def tryTimeModelUpdateOnAllParticipants(
```